doc/read-the-docs-site/tutorials/english-auction-example.rst

```diff
@@ -0,0 +1,344 @@
+
+This transaction will have two outputs:
+
+* One output is for the highest bidder, Charlie, who gets the NFT.
+* The second output goes to Alice who gets the highest bid.
+
+In the Close case, the script checks that these conditions are satisfied:
+
+1. The deadline has been reached.
+2. The winner gets the NFT.
+3. The auction owner gets the highest bid.
+
+When there are no bidders
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Finally, we need to consider the scenario in which nobody bids in the auction.
+Alice creates the auction, but the auction receives no bids.
+In this case, there must be a mechanism for Alice to retrieve her NFT.
+
+To address this possibility, Alice creates a transaction with the `Close` redeemer.
+However, because there is no bidder, the NFT doesn’t go to the highest bidder, but instead simply goes back to Alice.
+
+The logic in this case is slightly different.
+It will check that the NFT goes back to Alice.
+However, it doesn’t need to check the recipient because the transaction will be triggered by Alice and she can send the NFT wherever she wants.
+
+Diagram/graphic of how the smart contract works
+--------------------------------------------------
+
+*Assess what sort of diagrams would be most helpful.*
+
+Plutus Tx Validator Code
+-----------------------------
+
+With the on-chain code (the scripts from the EUTXO model), we have a script address in addition to public key addresses.
+Outputs can sit at such a script address, and if a transaction tries to consume such an output, the script is executed.
+The transaction is only valid if the script succeeds.
+
+If a node receives a new transaction, it validates it before accepting it into its mempool and eventually into a block.
+For each input of the transaction, if that input happens to be a script address, the corresponding script is executed.
+If the script does not succeed, the transaction is invalid and does not get completed.
+
+Plutus Core
+~~~~~~~~~~~~~~
+
+The programming language this script is expressed in is called (Untyped) Plutus Core, but you never write Plutus Core by hand.
+Instead, you write Plutus Tx (a subset of Haskell) and that gets compiled down to Plutus Core.
+Eventually there may be other high-level languages such as Solidity, C or Python that can compile down to Plutus Core.
+
+The task of a script is to say yes or no regarding whether a transaction can consume an output.
+
+Next, we'll go over the Plutus Tx code for the auction validator.
+The full source code can be found in `AuctionValidator.hs <https://github.com/input-output-hk/plutus/blob/master/doc/read-the-docs-site/tutorials/AuctionValidator.hs>`_.
+
+Data Types
+~~~~~~~~~~~~~~~~~~~~~~~
+
+First, let's define the following data types and instances for the validator:
+
+.. literalinclude:: AuctionValidator.hs
+   :start-after: BLOCK1
+   :end-before: BLOCK2
+
+The purpose of ``makeLift`` and ``unstableMakeIsData`` will be explained later.
```

Comment:
You could actually put them later in the program, then?

Reply:
The only option would be to put it at the beginning of the next block, but that feels awkward and distracting.